pkg/codegen/go/gen_program.go

```diff
@@ -36,30 +37,22 @@ type generator struct {
 	arrayHelpers        map[string]*promptToInputArrayHelper
 	isErrAssigned       bool
 	configCreated       bool
+	createResourceVar   bool // Assign resource to a new variable instead of _.
 }
 
-func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics, error) {
-	// Linearize the nodes into an order appropriate for procedural code generation.
-	nodes := pcl.Linearize(program)
+// GenerateProgramOption functions are used to configure optional generator behavior.
+type GenerateProgramOption func(g *generator)
 
-	packages, contexts := map[string]*schema.Package{}, map[string]map[string]*pkgContext{}
-	for _, pkg := range program.Packages() {
-		packages[pkg.Name], contexts[pkg.Name] = pkg, getPackages("tool", pkg)
+// CreateResourceVar enables the `createResourceVar` generator option.
+func CreateResourceVar() GenerateProgramOption {
```

Comment:
What do you think of a slightly more descriptive name for the option, e.g. `AssignResourcesToVariables` rather than `CreateResourceVar`?